game/views.py

```
@@ -1248,26 +1249,139 @@ def post(self, request, *args, **kwargs):
+def get_ip_lockout_key(ip):
+    """Get the cache key for IP lockout state."""
+    digest = hashlib.sha256(ip.encode('utf-8')).hexdigest()
+    return f'login_lockout:ip:{digest}'
+
+
+def increment_counter(key, timeout):
+    """Increment the value in the cache atomically if supported, or fall back gracefully."""
+    try:
+        if cache.add(key, 1, timeout=timeout):
+            return 1
+        else:
```

Ruff (0.15.15)

[warning] Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)